dfget/core/api/supernode_api.go

```diff
@@ -100,9 +107,14 @@ func (api *supernodeAPI) ReportPiece(node string, req *types.ReportPieceRequest)
 
 	url := fmt.Sprintf("%s://%s%s?%s",
 		api.Scheme, node, peerReportPiecePath, util.ParseQuery(req))
-
 	resp = new(types.BaseResponse)
-	e = api.get(url, resp)
+	if e = api.get(url, resp); e != nil {
+		logrus.Errorf("failed to report piece{taskid:%s,range:%s},err: %v", req.TaskID, req.PieceRange, e)
```

Comment:
Just return `error` in API implementation rather than logging them. Then the caller can choose to log the error themselves.

Reply:
dfgte haven't check all the err and response.so should I and a check ?  @lowzj

Reply:
https://github.com/dragonflyoss/Dragonfly/pull/682#discussion_r305193116 
I added a check before. and I think it is a good idea to add checks in supernode api